src/shared/mount-util.h

```diff
@@ -112,7 +112,11 @@ int mount_image_in_namespace(pid_t target, const char *propagate_path, const cha
 
 int make_mount_point(const char *path);
 
-int remount_idmap(const char *p, uid_t uid_shift, uid_t uid_range);
+typedef enum RemountIdmapFlags {
+        REMOUNT_IDMAP_HOST_ROOT = 1 << 0,
```

Comment:
it seems every call actually passes the flag, is it necessary at all? are there cases where we don't want to do this, or expect we won't?

Reply:
Good point. I added the flag explicitly though since it's not obvious, and it does create some attack surface if misused. Thus, callers should really explicitly request this, even though all our current callers *will* request this.

i.e. adding this extra uid mapping should always be done only after careful consideration that it is safe. And thus be opt-in instead of default.

Reply:
can you add a comment explaining this where the flag is defined then? It's not super obvious when it shouldn't be used, so an hint would make it easier to avoid mistakes

Reply:
done